feature/interests/src/main/java/com/google/samples/apps/nowinandroid/feature/interests/InterestsRoute.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.nowinandroid.feature.interests
+
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+
+@Composable
+internal fun InterestsRoute(
+    listState: LazyListState,
+    shouldShowTwoPane: Boolean,
+    onTopicClick: (String) -> Unit,
+    onBackClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    viewModel: InterestsViewModel = hiltViewModel(),
+) {
+    val interestUiState by viewModel.interestUiState.collectAsStateWithLifecycle()
+    val topicUiState by viewModel.topicUiState.collectAsStateWithLifecycle()
+
+    Row(modifier = modifier) {
+        if (shouldShowTwoPane || topicUiState == null) {
+            InterestsScreen(
+                uiState = interestUiState,
+                listState = listState,
+                followTopic = viewModel::followTopic,
+                onTopicClick = onTopicClick,
+                modifier = Modifier.weight(1f),
+            )
+        }
+
+        topicUiState?.let { state ->
```

Comment:
If there is no `topicUiState` since there is no `topicId`, we could consider adding some sort of placeholder here in the case where we want to show 2 panes.

Reply:
Yep, that sounds good